src/Middleware/StaticFiles/src/StaticFileOptions.cs

```diff
@@ -46,6 +47,14 @@ public StaticFileOptions(SharedOptions sharedOptions) : base(sharedOptions)
         /// </summary>
         public bool ServeUnknownFileTypes { get; set; }
 
+        /// <summary>
+        /// Indicates if files should be compressed for HTTPS requests. The default value is <see cref="HttpsCompressionMode.Compress"/>.
+        /// </summary>
+        /// <remarks>
+        /// Enabling compression on HTTPS requests for remotely manipulable content may expose security problems.
+        /// </remarks>
+        public HttpsCompressionMode HttpsCompression { get; set; } = HttpsCompressionMode.Compress;
```

Comment:
Do we compress for static files without having the response compression middleware?

Reply:
Only via IIS.